src/vs/workbench/parts/terminal/electron-browser/terminalInstance.ts

```diff
@@ -665,6 +667,8 @@ export class TerminalInstance implements ITerminalInstance {
 		const env = shell.env ? shell.env : TerminalInstance._cloneEnv(parentEnv);
 		env['PTYPID'] = process.pid.toString();
 		env['PTYSHELL'] = shell.executable;
+		env['TERM_PROGRAM'] = product.nameShort;
```

Comment:
Not sure if it would be better to use a constant name across all VSCode builds (insiders, stable, oss)?